src/vstest.console/Internal/ConsoleLogger.cs

```diff
@@ -707,9 +707,11 @@ private void TestRunCompleteHandler(object sender, TestRunCompleteEventArgs e)
                 var source = sd.Key;
                 var sourceSummary = new SourceSummary();
 
-                foreach (var result in sd.ToArray())
+                var results = sd.ToArray();
+                // duration of the whole source is the difference between the test that ended last and the one that started first
+                sourceSummary.Duration = !results.Any() ? TimeSpan.Zero : results.Max(r => r.EndTime) - results.Min(r => r.StartTime);
```

Comment:
```suggestion
                sourceSummary.Duration = results.Any() ? results.Max(r => r.EndTime) - results.Min(r => r.StartTime) : TimeSpan.Zero;
```

Reply:
why?

Reply:
Just a suggestion. More readable for me.

Reply:
It was set to auto-merge, and you approved, so it is not implemented :)